### DISCUSSION TITLE
Custom distros and desktop environments

### DISCUSSION THREAD

**Discussion by TheRealFanjin** [dnschneid/crouton#4393] [Q&A]
Is it possible to add your own distros or/and desktop environments?

**Comment by ghost** [comment]
Yes, you can make your own target by adding it to the `targets` folder, you can then run your own crouton version by running `main.sh` in the `installer` folder. You may need to move the crouton-master folder into `/usr/local/bin` due to new Chrome OS security (see #4026).

## Set by step
1. Download the crouton code by as a `.zip`.
2. Unzip the file so you have a `crouton-master` folder.
3. Move this file to `/usr/local/bin/`.
4. Make a copy of one of the target files and edit it to install your packages.

I like to edit it on a GitHub fork and edit there, then you can download it and copy it (using `sudo` as that will overwrite the old one), this will also save your code in case you mess something up and need to factory reset.

**Reply by TheRealFanjin** [reply]
Thanks for the answer! So for example, if I wanted to install Fedora what packages would I need and where would I get them?

**Reply by ghost** [reply]
For this you would need to create a custom release... not so much a desktop environment. I haven't tried it before but you could, probably, do it.

It appears to me that they are stored in the `installer` folder. If you made a copy of one of the already made ones (`debian`, `kali`, or `ubuntu`) You can modify the `defaults` file to have the Fedora mirrors and the `releases` file to have the Fedora releases then you should have a working Fedora install, although you would need to modify the files using `apt` or `apt-get` to have the fedora package manger, using detection of the release so the Debain, Ubuntu, and Kali releases still work. 

It would take a lot of time but it would, in the end, work.

**Reply by marauder316** [reply]
> It would take a lot of time but it would, in the end, work.

How stable would the new chroots be?

**Reply by ghost** [reply]
Well it depends on how stable they are made. You would need to add support for things here and there, that if you skipped could lead to problems that would lead it to be "unstable".

**Comment by marauder316** [comment]
OK, thanks

**Comment by DrDoktor** [comment]
since this is about custom distros and crouton is it possible for this [distro](https://dee.su/liberte) to be used

**Reply by marauder316** [reply]
I assume it would be, but I think the Liberté devs were clear when they mentioned potential instability due to age (last release was in 2012)

For context:
```
PLEASE NOTE: Latest release of Liberté Linux is old, and you should use it only if you understand the security implications.
A new release of Liberté, if created, will have significantly different usage objectives.
```

**Reply by ghost** [reply]
I agree with @marauder316, you should be able to get it to run on a Chromebook but I don't think app, or driver, support is going to be amazing.

**Comment by llvee** [comment]
@dnschneid Has crouton ever been considered as a security solution or developed in that way? I haven't realized the use case for Crouton in that way. Wondering if you might be able to network in someone on the Google ChromeOS team since I am interested in contributing to security solutions for more secure development environments on ChromeOS & what your thoughts are regarding the above question.

**Reply by dnschneid** [reply]
If you want a secure Linux dev environment, crostini is a lot more secure than crouton is.  I believe there's also ways you can tap into the container manager and run your own containers inside the VM.

crouton mainly just tries to minimize how much it breaks Chromium OS's security model, but inherently it has root access to the full system.

**Reply by llvee** [reply]
Thank you for sharing about the differences between Crostini, Crouton. 

I have much more progress to make with my own solutions.
